lite2/store/db/db.go

```diff
@@ -38,20 +38,21 @@ func (s *dbs) SaveSignedHeaderAndNextValidatorSet(sh *types.SignedHeader, valSet
 		panic("negative or zero height")
 	}
 
-	// TODO: batch
-	bz, err := s.cdc.MarshalBinaryLengthPrefixed(sh)
+	shBz, err := s.cdc.MarshalBinaryLengthPrefixed(sh)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "marshalling header")
 	}
-	s.db.Set(s.shKey(sh.Height), bz)
-
-	bz, err = s.cdc.MarshalBinaryLengthPrefixed(valSet)
+	valSetBz, err := s.cdc.MarshalBinaryLengthPrefixed(valSet)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "marshalling validator set")
 	}
-	s.db.Set(s.vsKey(sh.Height+1), bz)
 
-	return nil
+	b := s.db.NewBatch()
+	b.Set(s.shKey(sh.Height), shBz)
+	b.Set(s.vsKey(sh.Height+1), valSetBz)
+	err = b.WriteSync()
+	b.Close()
+	return err
```

Comment:
nit: wrap error so IF this fails we can pinpoint it a bit easier. if this were to fail there would be a larger problem at hand I believe.